komorebi/src/window_manager.rs

```diff
@@ -1454,6 +1456,11 @@ impl WindowManager {
             }
         }
 
+        let fl_windows = &mut self.focused_workspace_mut()?.floating_windows;
+        if let Some(idx) = fl_windows.elements().iter().position(|w| w.is_focused()) {
+            fl_windows.focus(idx);
+        }
+
```

Comment:
This alone is not sufficient. It does not update the indices when the workspace is empty, so it should be used together with `workspace.floating_windows.focus(idx)` in the `WindowManagerEvent::FocusChange(_, window)` handler. However, if you remove this code and leave only `workspace.floating_windows.focus(idx)` in `WindowManagerEvent::FocusChange(_, window)`, it won’t work immediately after making two windows floating one after another (and I don't know why).